### PR TITLE
Added session manager

### PR DESCRIPTION
**Pull request by tanushrigoel**
I have added a utility tool for developers which help them manage their productivity better. This is my first time creating a pull request please guide me incase of any error or anything

### CODE REVIEW

**Review by FerdausPolok — CHANGES_REQUESTED** [review]
Hello @tanushrigoel! Thanks for your contribution here! Would be great if you could add a `readme.md` which will help others to better understand your session manager. Try to add basic info, how to use, etc. Thanks.

**Comment by tanushrigoel** [comment]
Added readme.md file. Please check now.

**Comment by tanushrigoel** [comment]
Please add hacktoberfest-accepted labels

File: applications-code-snippets/javascript-session-manager/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# YouTube Session Manager
+
+A browser extension designed to help users manage their YouTube time by setting a session timer. Once the set time expires, the YouTube tab closes automatically, encouraging productive and time-conscious browsing.
+
+## Features
+
+- **Custom Timer**: Set a personalized time limit for your YouTube session.
+- **Auto Tab Close**: Automatically closes the YouTube tab once the timer runs out.
+- **Simple Interface**: User-friendly design for seamless interaction.
```
Comment (FerdausPolok) [inline applications-code-snippets/javascript-session-manager/README.md]: Can we add another header like

## How to run it?

File: applications-code-snippets/javascript-session-manager/background.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  chrome.alarms.onAlarm.addListener((alarm) => {
+    if (alarm.name === 'closeTab') {
+      chrome.tabs.query({ active: true, currentWindow: true }, (tabs) => {
+        if (tabs[0]) {
+          chrome.tabs.remove(tabs[0].id);
+        }
+      });
+    }
+  });
+
```
Comment (FerdausPolok) [inline applications-code-snippets/javascript-session-manager/background.js]:
```suggestion
  });

```

File: applications-code-snippets/javascript-session-manager/manifest.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      "service_worker": "background.js"
+    },
+    "action": {
+      "default_popup": "popup.html"
+    },
+    "host_permissions": [
+      "https://*.youtube.com/*"
+    ]
+  }
+
```
Comment (FerdausPolok) [inline applications-code-snippets/javascript-session-manager/manifest.json]:
```suggestion
  }

```

File: applications-code-snippets/javascript-session-manager/styles.css [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    font-size: 1.2em;
+    border-radius: 5px;
+    cursor: pointer;
+    transition: background-color 0.3s;
+  }
+  
+  .start-button:hover {
+    background-color: #218838;
+  }
+
```
Comment (FerdausPolok) [inline applications-code-snippets/javascript-session-manager/styles.css]:
```suggestion
  }

```

**Review by FerdausPolok — CHANGES_REQUESTED** [review]
I tried to local run your project with VSCode's Go Live feature which runns the `popup.html` into a spesific `port` of my `localhost`. (IDK if that the proper way to run or not, please correct me if I'm wrong. )

Then when I set a time like 1-2 min and then hit `Start` button it shows below error into console and nothing happens:

```
popup.js:45 Uncaught TypeError: Cannot read properties of undefined (reading 'local')
    at HTMLButtonElement.<anonymous> (popup.js:45:18)
```

Could you please look into it first? Thanks.

**Comment by tanushrigoel** [comment]
Please check now

**Review by FerdausPolok — CHANGES_REQUESTED** [review]
Yeap @tanushrigoel it's working now! Below are some requested improvement point. Please have a look and ask questions if you have any confussions. 

1. Could you please now remove the older one? Right now I can see there are 2 folders.

2. Also it would be great if you can improve the user experience by starting the timer right after clickinging Start button. Cause when a user clicks on Start button nothing happen, so it's pretty confusion to know if my timer has started or not. Then once I click somewhere else to close the extention and click on it again then I see my timer.  So basically what I'm asking is that, we see the timer imediately reducing time by sec right after user clicks on start. Can you do that?

3. Lastly, it would be great if you can format all the files inside `applications-code-snippets\javascript-youtube-session-manager\` using [Prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)

**Review by FerdausPolok — COMMENTED** [review]
Thanks you for your work so far.

**Comment by tanushrigoel** [comment]
Thank you for your patience. I have done the necessary changes. Please check now @FerdausPolok

**Review by FerdausPolok — COMMENTED** [review]
Cool! Great work @tanushrigoel! 🚀

Btw we can enhance the user experience by addressing this issue:

When a timer is set (e.g., 1 minute), the + and - buttons can still be clicked, but their functionality is missing. The time appears to increase or decrease momentarily, but then resets to the last counted time.

To fix this, we could do a few things:

1. **Add functionality to adjust the time:** Allow the time to increase or decrease while the timer is running. However, if the timer is below 1 minute (e.g., 50 seconds), the - button should be disabled since the time can't go below 0.

3. **Disable the + and - buttons after starting the timer:** Once the timer starts, disable both buttons. Add a "Reset" button next to "Start" so the user can cancel the current timer and set a new one if needed.

If you'd like to include these improvements in the current PR, that would be great. But if you prefer to create a new issue and submit another PR, that's fine too. Let me know how you'd like to proceed. Thanks!

**Comment by tanushrigoel** [comment]
I will just update this pr and submit it after updating if it's fine

**Review by FerdausPolok — APPROVED** [review]

**Comment by FerdausPolok** [comment]
> I will just update this pr and submit it after updating if it's fine

Okay then feel free to start further working by creating an issue first. Thanks.

**Pull request merged**